src/lang/modifyAst.ts

```diff
@@ -163,39 +163,51 @@ export function addSketchTo(
   }
 }
 
+/**
+ * Given a target name and an AST,
+ * return a unique name that is not already used as
+ * a constant or tag.
+ */
 export function findUniqueName(
-  ast: Program | string,
+  ast: Program,
   name: string,
   pad = 3,
   index = 1
 ): string {
-  let searchStr: string = typeof ast === 'string' ? ast : JSON.stringify(ast)
+  // Searches for partial TagDeclarator or VariableDeclarator strings
+  const searchStrings = (n: string) => [
+    `{"type":"TagDeclarator","value":"${n}"`,
+    `{"type":"VariableDeclarator","id":{"type":"Identifier","name":"${n}"`,
```

Comment:
What guarantees that JS objects' keys are serialized in this order? I don't think this is safe.